android/app/src/androidTest/java/org/fossasia/openevent/DateTest.java

```diff
@@ -0,0 +1,362 @@
+package org.fossasia.openevent;
+
+import org.fossasia.openevent.utils.ISO8601Date;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Created by arpitdec5 on 02-06-2017.
+ */
+
+public class DateTest {
+
+    @Test
+    public void shouldConvertSingaporeDateStringToUTCDateString() {
```

Comment:
Name format is correct but should not be particular like singapore or kolkata

It should be like shouldConvertLocalToInternational and then contain several tests on same string for different locales using same method.

For example, you are testing getTimeZoneDateStringFromString, so the test will only contain one date to be tested. And it will be tested on different timezones using the same method, so that if this particular test fails, we are sure that this particular feature of this method, namely changing the local date to international isn't working. All tests are good, just a little rearranging will do

One more thing. Keep the several date testing in one method for each function, like here you are using 4 strings, test them only once for a particular locale and this test will tell that the date conversion is correct. And when this test along with the above specified test (of timezone conversion) passes, we will know that both these functions of that method are working well

So, one containing 4 strings and testing on single locale - date conversion is correct
One containing single string and converting them to different locale or single string converting from different locale to local - timezone conversion is correct

This way each method will have 3 symmetrical tests which if failing will tell us that which component of the function is faulty